internal/core/app_status_report.go

```diff
@@ -0,0 +1,340 @@
+package core
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/zclconf/go-cty/cty"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/go-argmapper"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl/v2"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	sdk "github.com/hashicorp/waypoint-plugin-sdk/proto/gen"
+	"github.com/hashicorp/waypoint/internal/config"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+func (a *App) DeploymentStatusReport(
+	ctx context.Context,
+	deployTarget *pb.Deployment,
+) (*pb.StatusReport, *sdk.StatusReport, error) {
+	var evalCtx hcl.EvalContext
+	// Load the deployment variables context
+	if err := a.deployStatusReportEvalContext(ctx, deployTarget, &evalCtx); err != nil {
+		return nil, nil, err
+	}
+
+	// Load variables from deploy
+	hclCtx := evalCtx.NewChild()
+	if _, err := a.deployEvalContext(ctx, hclCtx); err != nil {
+		return nil, nil, err
+	}
+
+	c, err := a.createStatusReporter(ctx, hclCtx, component.PlatformType)
+	if status.Code(err) == codes.Unimplemented {
+		c = nil
+		err = nil
+	}
```

Comment:
Should this just return an empty status report right away? It looks like the code below here doesn't deal with `c` being nil.